package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,121 @@
+        must be a list of one or more AtomGroups containing 5 atoms in the
+        correct order (i.e. C-N-CA-C-N)
+
+    Returns
+    -------
+    angles : numpy.ndarray
+        An array of time steps which contain (phi,psi) for all AtomGroups.
+    """
+
+    dihedrals = [atomgroup.dihedral for atomgroup in atomgroups]
+    angles = [dih.value() for dih in dihedrals]
+
+    return angles
+
+
+class Ramachandran(AnalysisBase):
+    """Calculate phi and psi dihedral angles of specified residues.
+
+    Note
+    ----
+    Run the analysis with :meth:`Ramachandran.run()`, which stores the results
+    in the array :attr:`Ramachandran.angles`. A axes object can be obtained
+    with :meth: `Ramachandran.run().plot()`.
+
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the first or last residue
+    then a warning will be raised, and the final array of angles will not
+    include those residues.
+
+    """
+    def __init__(self, atomgroup, **kwargs):
+        r"""Parameters
+        ----------
+        atomgroup : Atomgroup
+            atoms for residues for which phi and psi are calculated
+        start : int, optional
+            starting frame, default None becomes 0.
+        stop : int, optional
+            Frame index to stop analysis. Default: None becomes
+            n_frames. Iteration stops *before* this frame number,
+            which means that the trajectory would be read until the end.
+        step : int, optional
+            step between frames, default None becomes 1.
+
+        """
+        super(Ramachandran, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+
+    def _prepare(self):
+        self.residues = self.atomgroup.residues
+        res_min = np.min(self.atomgroup.universe.select_atoms("protein").residues)
+        res_max = np.max(self.atomgroup.universe.select_atoms("protein").residues)
+        if any([(residue < res_min or residue > res_max) for residue in self.residues]):
+            raise IndexError("Selection exceeds protein length")
+        elif any([residue == (res_min or res_max) for residue in self.residues]):
+            warnings.warn("Cannot determine phi and psi angles for the first or last residues")
+
+        self.phi_atoms = [residue.phi_selection() for residue in self.residues
+                          if res_min < residue < res_max]
+        self.psi_atoms = [residue.psi_selection() for residue in self.residues
+                          if res_min < residue < res_max]
+
+        self.angles = []
+
+    def _single_frame(self):
+        self.phi_angles = dihedral_calc(self.phi_atoms)
+        self.psi_angles = dihedral_calc(self.psi_atoms)
+        self.angles.append((self.phi_angles,self.psi_angles))
+
+    def _conclude(self):
+        self.angles = np.array(self.angles)
+
+    def plot(self, ax=None, color='k', marker='s', title=None):
```

Comment:
this is better suited as a free function that takes a Ramachandran class as first argument. Also you should allow for arbitrary kwargs and then call the plotting function `ax.plot(...., **kwargs)`. This allows the user to select his own styling and better integrate the plots.

Reply:
None of the other analysis classes are using plot from an external function. And at least I think it makes sense to keep it as a method, along the lines of "the method knows how to plot the data". Most people will initially use it to get some quick results, the perhaps use it to plot into their own `ax` (that's important), and then finally just reverse-engineer the code to see what they need for their own plotting. I feel it is useful to have `Ramachandran.plot()` as it immediately tells you what you can do.

Reply:
I think I would omit all styling kwargs (color, marker, title) and just pass the `**kwargs` to plot. Then let the user do the styling, should work just fine with scatter plot as @kain88-de is suggesting.